src/screens/Saved.js

```diff
@@ -42,72 +44,72 @@ export default class Saved extends Component<Props> {
                 style={{
                   paddingTop: 64,
                   marginBottom: 20,
-                  opacity: .8,
+                  opacity: 0.8
                 }}
               />
             </TouchableOpacity>
           </View>
-          <SubHeading>
-            {events.length} events saved
-          </SubHeading>
-          {
-            (pastEvents.length > 0) &&
-              <Fragment>
-                <TouchableOpacity
-                  onPress={() => {
-                    this.setState({ showPastEvents: !this.state.showPastEvents});
-                  }}
-                >
-                  <Button
-                    text={`${this.state.showPastEvents ? 'Hide' : 'Show'} ${pastEvents.length} past event${pastEvents.length > 1 ? 's' : ''}`}
-                    style={styles.showPastEventsButton}
-                  />
-                </TouchableOpacity>
-              </Fragment>
-          }
-          {
-            (pastEvents.length > 0) && (this.state.showPastEvents) &&
+          {/* Details how many events were saved by the user. */}
+          <SubHeading>{events.length} events saved</SubHeading>
+          {/* Logic and components that allow the user to hide or display past events. */
+          pastEvents.length > 0 && (
             <Fragment>
-              <EventsList
-                events={pastEvents}
-                eventManager={eventManager}
-              />
+              <TouchableOpacity
+                onPress={() => {
+                  this.setState({ showPastEvents: !this.state.showPastEvents });
+                }}
+              >
+                <Button
+                  text={`${this.state.showPastEvents ? "Hide" : "Show"} ${
+                    pastEvents.length
+                  } past event${pastEvents.length > 1 ? "s" : ""}`}
+                  style={styles.showPastEventsButton}
+                />
+              </TouchableOpacity>
             </Fragment>
-          }
-          {
-            (upcomingEvents.length > 0) &&
+          )}
+          {/* The list of past events, only shown if user selected this option */
+          pastEvents.length > 0 && this.state.showPastEvents && (
             <Fragment>
-              <EventsList
-                events={upcomingEvents}
-                eventManager={eventManager}
-              />
+              <EventsList events={pastEvents} eventManager={eventManager} />
+            </Fragment>
+          )}
+          {/* List of upcoming events saved by the user. */
+          upcomingEvents.length > 0 && (
+            <Fragment>
+              <EventsList events={upcomingEvents} eventManager={eventManager} />
             </Fragment>
-          }
+          )}
         </PadContainer>
-
       </ScrollView>
     );
   }
 }
 
+/**
+ * List of events that the user has saved.
+ */
 class EventsList extends Component<Props> {
   constructor(props) {
     super(props);
   }
 
   componentWillUnmount() {
```

Comment:
Maybe add a comment explaining why we are using the componentWillMount vs componentDidMount lifecycle methods, that way people who may be unfamiliar with those methods will understand what the difference is